Mathlib/NumberTheory/ModularForms/LevelOne.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import Mathlib.NumberTheory.Modular
+import Mathlib.NumberTheory.ModularForms.SlashInvariantForms
+import Mathlib.NumberTheory.ModularForms.CongruenceSubgroups
+
+/-!
+# Level one modular forms
+
+This file contains results specific to modular forms of level one, ie. modular forms for `SL(2, ℤ)`.
+
+TODO: Add finite dimisionality of these spaces of modular forms.
+
+-/
+
+open UpperHalfPlane ModularGroup SlashInvariantForm ModularForm CongruenceSubgroup Complex
+local notation "SL(" n ", " R ")" => Matrix.SpecialLinearGroup (Fin n) R
+local notation  "Γ(" n ")"  => CongruenceSubgroup.Gamma n
```

Comment:
These local notations are getting reused in many places, they should be promoted to `scoped notation` which can be re-opened when needed.

Reply:
I scoped them in CongruenceSubgroup namespace, but I wanted to try and do it for SlashInvariantForms, but I couldn't seem to do it for SL since the notation was also local (and changing all the notation in the top half of the file seemed to annoying to do). Is there a way to forget the notation halfway down a file?

Reply:
Let me take a look.